pkg/trace/trace.go

```diff
@@ -0,0 +1,217 @@
+	rootTrace["startTimeMillis"] = time.Now().UnixNano() / int64(time.Millisecond)
+	rootTrace["flags"] = 1
+	rootTrace["operationName"] = "HTTP GET"
+	rootTrace["duration"] = fastrand.Uint32n(1_000)
+	rootTrace["tags"] = generateTagBody(10)
+	rootTrace["process"] = generateProcessTagsBody(10)
+	rootTrace["references"] = make([]map[string]interface{},0)
+	return rootTrace
+}
+
+func WriteToServiceFile(serviceAndOperationJson map[string]string, w *bufio.Writer) {
+	bytes, _ := json.Marshal(serviceAndOperationJson)
+	_, err1 := w.Write(bytes)
+	if err1 != nil {
+		log.Fatal(err1)
+	}
+	_, err2 := w.WriteString("\n")
+	if err2 != nil {
+		log.Fatal(err2)
+	}
+}
+
+func WriteToFile(span map[string]interface{}, w *bufio.Writer) {
+	bytes, _ := json.Marshal(span)
+	_, err1 := w.Write(bytes)
+	if err1 != nil {
+		log.Fatal(err1)
+	}
+	_, err2 := w.WriteString("\n")
+	if err2 != nil {
+		log.Fatal(err2)
+	}
+}
+
+func getServiceAndOperationObject(operationName, processTags interface{}) map[string]string {
+	serviceAndOperationObject := make(map[string]string)
+	if opName, ok := operationName.(string); ok {
+		serviceAndOperationObject["operationName"] = opName
+	}
+	if serviceName, ok := processTags.(map[string]interface{})["serviceName"].(string); ok {
+		serviceAndOperationObject["serviceName"] = serviceName
+	}
+	return serviceAndOperationObject
+}
+
+func generateChildSpan(traceId string, spanCounter int,prevSpanId string) (map[string]interface{},string) {
+	span := make(map[string]interface{})
+	spanId := randomHex(8)
+	span["spanID"] = spanId
+	span["startTime"] = time.Now().UnixNano() / int64(time.Microsecond)
+	span["startTimeMillis"] = time.Now().UnixNano() / int64(time.Millisecond)
+	span["flags"] = 1
+	span["operationName"] = "HTTP GET"
+	span["duration"] = fastrand.Uint32n(1_000)
+	span["tags"] = generateTagBody(10)
+	span["process"] = generateProcessTagsBody(10)
+	span["references"] = generateReferenceBody(traceId, spanId, spanCounter,prevSpanId)
+	return span,spanId
+}
+
+func generateReferenceBody(traceId string, spanId string, spanCounter int,prevSpanId string) []map[string]interface{} {
+	references := make([]map[string]interface{}, 1)
+	reference := make(map[string]interface{})
+	if spanCounter == 0 {
+		reference["refType"] = "CHILD_OF"
+		reference["spanID"] = traceId
+	} else {
+		reference["refType"] = "FOLLOWS_FROM"
+		reference["spanID"] = prevSpanId
+	}
+	reference["traceID"] = traceId
+	references[0] = reference
+	return references
+}
+
+func generateProcessTagsBody(n int) map[string]interface{} {
+	processBody := make(map[string]interface{})
+	randomServiceId := fastrand.Uint32n(10)
+	processBody["serviceName"] = fmt.Sprintf("service-%d", randomServiceId)
+	listOfTags := make([]map[string]interface{}, 3)
+	tagsCounter := 0
+	for tagsCounter < 3 {
+		tags := make(map[string]interface{})
+		randomNodeId := fastrand.Uint32n(2_000)
+		randomPodId := fastrand.Uint32n(20_000)
+		randomUserId := fastrand.Uint32n(2_000_000)
```

Comment:
does this need to be in a loop if `tagsCounter == 3` at the end of the first iteration?